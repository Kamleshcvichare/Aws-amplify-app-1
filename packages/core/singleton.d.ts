export * from './lib-esm/Singleton';