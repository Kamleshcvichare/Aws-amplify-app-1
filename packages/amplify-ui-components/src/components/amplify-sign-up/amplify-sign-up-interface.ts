export interface AmplifySignUpFormFooterProps {
  submitButtonText: string;
  haveAcccountText: string;
  signInText: string;
  overrideStyle?: boolean;
  handleAuthStateChange?: any;
}

export interface AmplifySignUpAttributes {
  username: string;
  password: string;
  attributes?: {
    email?: string;
    phone_number?: string;
  };
}
