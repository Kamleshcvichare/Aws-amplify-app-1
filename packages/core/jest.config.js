module.exports = {
	...require('../../jest.config'),
	coverageThreshold: {
		global: {
			branches: 74, // todo(ashwinkumar6): temp, add tests and bump back
			functions: 72,
			lines: 87, // todo(ashwinkumar6): temp, add tests and bump back
			statements: 88,
		},
	},
};
