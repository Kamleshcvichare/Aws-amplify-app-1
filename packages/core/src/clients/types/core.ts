/**
 * General None HTTP-specific request interface
 */
export interface Request {
	url: URL;
	body?: unknown;
}

export interface Response {
	body: unknown;
}

export interface TransferHandler<
	Input extends Request,
	Output extends Response,
	TransferOptions
> {
	(request: Input, options: TransferOptions): Promise<Output>;
}

/**
 * A slimmed down version of the AWS SDK v3 middleware handler, only handling instantiated requests
 */
export type MiddlewareHandler<
	Input extends Request,
	Output extends Response,
	MiddlewareOptions
> = (request: Input, options: MiddlewareOptions) => Promise<Output>;

export type MiddlewareContext = Record<string, unknown>;

/**
 * A slimmed down version of the AWS SDK v3 middleware, only handling tasks after Serde.
 */
export type Middleware<
	Input extends Request,
	Output extends Response,
	MiddlewareOptions
> = (
	next: MiddlewareHandler<Input, Output, MiddlewareOptions>,
	context: MiddlewareContext
) => MiddlewareHandler<Input, Output, MiddlewareOptions>;
