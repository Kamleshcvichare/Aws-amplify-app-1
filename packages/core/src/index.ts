// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
// SPDX-License-Identifier: Apache-2.0

/*
This file maps top-level exports from `@aws-amplify/core`. These are intended to be potentially customer-facing exports.
*/
// Hub exports
export { Hub } from './Hub';
export { HubCapsule, HubCallback, HubPayload } from './Hub/types';

// Singleton exports
export {
	TokenProvider,
	AuthTokens,
	FetchAuthSessionOptions,
	AWSCredentialsAndIdentityIdProvider,
	AWSCredentialsAndIdentityId,
	Identity,
	OAuthConfig,
	CognitoUserPoolConfig,
} from './singleton/Auth/types';
export {
	AuthConfig,
	AuthUserPoolConfig,
	AuthUserPoolAndIdentityPoolConfig,
	APIConfig,
	CacheConfig,
	StorageAccessLevel,
	StorageConfig,
	GetCredentialsOptions,
	ResourcesConfig,
	LibraryOptions,
	AnalyticsConfig,
	GeoConfig,
} from './singleton/types';
export {
	Amplify,
	fetchAuthSession,
	AmplifyClass as AmplifyClassV6,
	clearCredentials,
} from './singleton';

// AWSClients exports
export {
	getCredentialsForIdentity,
	getId,
	GetCredentialsForIdentityInput,
	GetCredentialsForIdentityOutput,
} from './awsClients/cognitoIdentity';

// Amplify-wide constructs
export { UserProfile } from './types';

// Storage helpers
export {
	CookieStorage,
	defaultStorage,
	sessionStorage,
	sharedInMemoryStorage,
} from './storage';
export { KeyValueStorageInterface } from './types';

// Cache exports
export { Cache } from './Cache';

// Internationalization utilities
export { I18n } from './I18n';

// Logging utilities
export { ConsoleLogger } from './Logger';
