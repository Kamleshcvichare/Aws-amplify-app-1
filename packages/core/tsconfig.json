{
	"extends": "../../tsconfig.json",
	"compilerOptions": {
		"baseUrl": ".",
		"importHelpers": false,
		"paths": {
			"~/*": ["./*"]
		}
	},
	"include": ["./src", "__tests__"]
}
