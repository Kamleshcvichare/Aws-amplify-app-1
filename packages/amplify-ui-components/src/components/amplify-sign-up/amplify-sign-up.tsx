import { Component, Prop, h, State } from '@stencil/core';
import { FormFieldTypes } from '../../components/amplify-auth-fields/amplify-auth-fields-interface';
import {
  SIGN_UP_HEADER_TEXT,
  SIGN_UP_SUBMIT_BUTTON_TEXT,
  HAVE_ACCOUNT_TEXT,
  SIGN_IN_TEXT,
  SIGN_UP_USERNAME_PLACEHOLDER,
  SIGN_UP_PASSWORD_PLACEHOLDER,
  SIGN_UP_PHONE_NUMBER_LABEL,
  SIGN_UP_PHONE_NUMBER_PLACEHOLDER,
} from '../../common/constants';
import { AmplifySignUpFormFooter } from './amplify-sign-up-form-footer';
import { AuthState } from '../../common/types/auth-types';
import { AmplifySignUpAttributes } from './amplify-sign-up-interface';

import { Auth } from '@aws-amplify/auth';

@Component({
  tag: 'amplify-sign-up',
  shadow: false,
})
export class AmplifySignUp {
  /** Fires when sign up form is submitted */
  @Prop() handleSubmit: (submitEvent: Event) => void = () => this.signUp();
  /** Engages when invalid actions occur, such as missing field, etc. */
  @Prop() validationErrors: string;
  /** Used for header text in sign up component */
  @Prop() headerText: string = SIGN_UP_HEADER_TEXT;
  /** Used for the submit button text in sign up component */
  @Prop() submitButtonText: string = SIGN_UP_SUBMIT_BUTTON_TEXT;
  /** Used for the submit button text in sign up component */
  @Prop() haveAccountText: string = HAVE_ACCOUNT_TEXT;
  /** Used for the submit button text in sign up component */
  @Prop() signInText: string = SIGN_IN_TEXT;
  /** (Optional) Overrides default styling */
  @Prop() overrideStyle: boolean = false;
  /**
   * Form fields allows you to utilize our pre-built components such as username field, code field, password field, email field, etc.
   * by passing an array of strings that you would like the order of the form to be in. If you need more customization, such as changing
   * text for a label or adjust a placeholder, you can follow the structure below in order to do just that.
   * ```
   * [
   *  {
   *    type: 'username'|'password'|'email'|'code'|'default',
   *    label: string,
   *    placeholder: string,
   *    hint: string | Functional Component | null,
   *    required: boolean
   *  }
   * ]
   * ```
   */
  @Prop() formFields: FormFieldTypes | string[];
  /** Passed from the Authenticatior component in order to change Authentication state
   * e.g. SignIn -> 'Create Account' link -> SignUp
   */
  @Prop() handleAuthStateChange: (nextAuthState: AuthState, data?: object) => void;

  @State() username: string;
  @State() password: string;
  @State() email: string;
  @State() phone_number: string;

  componentWillLoad() {
    this.formFields = [
      {
        type: 'username',
        placeholder: SIGN_UP_USERNAME_PLACEHOLDER,
        required: true,
        handleInputChange: event => this.handleUsernameChange(event),
      },
      {
        type: 'password',
        placeholder: SIGN_UP_PASSWORD_PLACEHOLDER,
        required: true,
        handleInputChange: event => this.handlePasswordChange(event),
      },
      {
        type: 'email',
        required: true,
        handleInputChange: event => this.handleEmailChange(event),
      },
      {
        type: 'phone',
        label: SIGN_UP_PHONE_NUMBER_LABEL,
        placeholder: SIGN_UP_PHONE_NUMBER_PLACEHOLDER,
        required: true,
        handleInputChange: event => this.handlePhoneNumberChange(event),
      },
    ];
  }

  handleUsernameChange(event) {
    this.username = event.target.value;
  }

  handlePasswordChange(event) {
    this.password = event.target.value;
  }

  handleEmailChange(event) {
    this.email = event.target.value;
  }

  handlePhoneNumberChange(event) {
    this.phone_number = event.target.value;
  }

  // TODO: Add validation
  // TODO: Prefix
  async signUp() {
    if (event) {
      event.preventDefault();
    }
    if (!Auth || typeof Auth.signUp !== 'function') {
      throw new Error('No Auth module found, please ensure @aws-amplify/auth is imported');
    }

    const signUpAttrs: AmplifySignUpAttributes = {
      username: this.username,
      password: this.password,
      attributes: {
        email: this.email,
        phone_number: this.phone_number,
      },
    };

    try {
      const user = await Auth.signUp(signUpAttrs);

      this.handleAuthStateChange(AuthState.ConfirmSignUp, user);
    } catch (error) {
      throw new Error(error);
    }
  }

  render() {
    return (
      <amplify-form-section
        headerText={this.headerText}
        overrideStyle={this.overrideStyle}
        handleSubmit={this.handleSubmit}
      >
        <amplify-auth-fields formFields={this.formFields} />
        <div slot="amplify-form-section-footer">
          <AmplifySignUpFormFooter
            submitButtonText={this.submitButtonText}
            haveAcccountText={this.haveAccountText}
            signInText={this.signInText}
            handleAuthStateChange={this.handleAuthStateChange}
          />
        </div>
      </amplify-form-section>
    );
  }
}
