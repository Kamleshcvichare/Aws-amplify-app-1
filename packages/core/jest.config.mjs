import { getJestConfig } from '../../jest/getJestConfig.mjs';
import tsconfig from './tsconfig.json' assert { type: 'json' };

const jestConfig = getJestConfig(tsconfig.compilerOptions, {
	coverageThreshold: {
		global: {
			branches: 75,
			functions: 72,
			lines: 88,
			statements: 88,
		},
	},
});

export default jestConfig;
