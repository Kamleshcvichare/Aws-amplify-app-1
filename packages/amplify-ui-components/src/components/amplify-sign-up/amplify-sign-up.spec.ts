import { newSpecPage } from '@stencil/core/testing';
import { AmplifySignUp } from './amplify-sign-up';

describe('amplify-sign-in spec:', () => {
  describe('Component logic ->', () => {
    let signUp;

    beforeEach(() => {
      signUp = new AmplifySignUp();
    });

    it('should render `handleSubmit` as undefined by default', () => {
      expect(signUp.handleSubmit).toBeUndefined();
    });

    it('should render `validationErrors` as undefined by default', () => {
      expect(signUp.validationErrors).toBeUndefined();
    });

    it('should render `headerText` to `Create a new account` by default', () => {
      expect(signUp.headerText).toEqual('Create a new account');
    });

    it('should render `submitButtonText` to `Create account` by default', () => {
      expect(signUp.submitButtonText).toEqual('Create account');
    });

    it('should render `overrideStyle` to false by default', () => {
      expect(signUp.overrideStyle).toBe(false);
    });
  });
  describe('Render logic ->', () => {
    it('should render a `sign in` form by default', async () => {
      const page = await newSpecPage({
        components: [AmplifySignUp],
        html: `<amplify-sign-up></amplify-sign-up>`,
      });

      expect(page.root).toMatchSnapshot();
    });
  });
});
