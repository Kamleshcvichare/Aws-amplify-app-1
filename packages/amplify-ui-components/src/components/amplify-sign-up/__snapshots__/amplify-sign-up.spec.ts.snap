// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`amplify-sign-up spec: Render logic -> should render a \`sign up\` form by default 1`] = `
<amplify-sign-up>
  <context-consumer></context-consumer>
</amplify-sign-up>
`;
